### PR TITLE
add github-api-gist.spec.js tests

### PR DESCRIPTION
**Pull request by renardete**

### CODE REVIEW

File: test/github-api-issue.spec.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    it('then the response should contain the name and id of the selected repository', async () => {
+      expect(response.status).toBe(statusCode.OK);
+      expect(response.body.name).toBe(apiWorkshopRepo.name);
+      expect(response.body.id).toBe(apiWorkshopRepo.id);
+    });
+  });
+
+
+  describe('', () => {
```
Comment (leonleo997) [inline test/github-api-issue.spec.js]: this should have a content

File: test/github-api-gist.spec.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      .set('User-Agent', 'agent')
+      .send(bodyGist);
+
+    responseResourceGetOriginal = await agent
+      .get(responsePost.body.url)
+      .auth('token', process.env.ACCESS_TOKEN)
+      .set('User-Agent', 'agent');
+
+    responseDelete = await agent
+      .del(responsePost.body.url)
```
Comment (leonleo997) [inline test/github-api-gist.spec.js]: this should be into a before all of the delete request describe

File: test/github-api-gist.spec.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    files: {
+      hello_world: {
+        content: 'class HelloWorld\n   def initialize(name)\n      @name = name.capitalize\n   end\n   def sayHi\n'
+      }
+    }
+  };
+
+  beforeAll(async () => {
+    responsePost = await agent
+      .post(`${urlBase}/gists`)
```
Comment (leonleo997) [inline test/github-api-gist.spec.js]: this should be into a before all of the post request describe

File: test/github-api-gist.spec.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      .auth('token', process.env.ACCESS_TOKEN)
+      .set('User-Agent', 'agent')
+      .send(bodyGist);
+
+    responseResourceGetOriginal = await agent
+      .get(responsePost.body.url)
```
Comment (leonleo997) [inline test/github-api-gist.spec.js]: this should be into a before all of the get request describe

File: test/github-api-gist.spec.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      .set('User-Agent', 'agent');
+
+    responseDelete = await agent
+      .del(responsePost.body.url)
+      .auth('token', process.env.ACCESS_TOKEN)
+      .set('User-Agent', 'agent');
+
+    try {
+      responseResourceGetrepeated = await agent
+        .get(responsePost.body.url)
```
Comment (leonleo997) [inline test/github-api-gist.spec.js]: this should be into a before all of the get request describe

File: test/github-api-issue.spec.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      title: 'A new issue '
+    };
+
+    const bodyPatch = {
+      body: 'new boddy for issue'
+    };
+
+    beforeAll(async () => {
+      responsePost = await agent
+        .post(`${urlBase}/repos/${userName}/${repoName}/issues`)
```
Comment (leonleo997) [inline test/github-api-issue.spec.js]: this should be into a before all of the post request describe

File: test/github-api-issue.spec.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+        .auth('token', process.env.ACCESS_TOKEN)
+        .set('User-Agent', 'agent')
+        .send(bodyPost);
+
+      responsePut = await agent
+        .patch(`${urlBase}/repos/${userName}/${repoName}/issues/${responsePost.body.number}`)
```
Comment (leonleo997) [inline test/github-api-issue.spec.js]: this should be into a before all of the patch request describe

File: test/github-api-gist.spec.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        .auth('token', process.env.ACCESS_TOKEN)
+        .set('User-Agent', 'agent');
+    } catch (err) {
+      responseResourceGetrepeated = err.response;
+    }
+  });
+
+  describe('when POST /gists resource is called', () => {
```
Comment (aperdomob) [inline test/github-api-gist.spec.js]: La logica de crear un gist deberia estar dentro de este describe

File: test/github-api-issue.spec.js (hunk 1 above)
Comment (aperdomob) [reply test/github-api-issue.spec.js]: agregar detalle

File: test/github-api-issue.spec.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        .auth('token', process.env.ACCESS_TOKEN)
+        .set('User-Agent', 'agent')
+        .send(bodyPatch);
+    });
+
+    describe(`when ${urlBase}/repos/${userName}/${repoName}/issues resource is called`, () => {
```
Comment (aperdomob) [inline test/github-api-issue.spec.js]: dentro de este describe deberia estar el llamado al resource

File: test/github-api-issue.spec.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+      it(`then the response should contain the issue title ${bodyPost.title} and body must be empty`, async () => {
+        expect(responsePost.status).toBe(statusCode.CREATED);
+        expect(responsePost.body.title).toBe(bodyPost.title);
+        expect(responsePost.body.body).toEqual(emptyString);
+      });
+    });
+
+    describe(`when ${urlBase}/repos/${userName}/${repoName}/issues/:numberIssue resource is called`, () => {
```
Comment (aperdomob) [inline test/github-api-issue.spec.js]: dentro de este describe deberia estar el llamado al resource

File: test/github-api-gist.spec.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+      } catch (err) {
+        responseResourceGetrepeated = err.response;
+      }
+    });
+
+    it('then the response should contain status=204 and body must be empty', () => {
+      expect(responseDelete.status).toBe(statusCode.NO_CONTENT);
+    });
+
+    it('then the resource must not exists', () => {
```
Comment (leonleo997) [inline test/github-api-gist.spec.js]: use and instead of then

File: test/github-api-issue.spec.js [hunk 12]
```diff
@@ -10,14 +11,9 @@ describe(`Given ${urlBase} domain is up`, () => {
...
-      response = await agent
-        .get(loggedUser)
-        .auth('token', process.env.ACCESS_TOKEN)
-        .set('User-Agent', 'agent');
-      response = await agent
-        .get(response.body.repos_url)
-        .auth('token', process.env.ACCESS_TOKEN)
-        .set('User-Agent', 'agent');
+      response = await httpUtils.authGetSync(loggedUser);
+
```
Comment (leonleo997) [inline test/github-api-issue.spec.js]: delete whitespace

**Comment by leonleo997** [comment]
Future task: create put, patch, delete, ... methods into http-utils class

**Pull request merged**